src/ring_theory/norm.lean

```diff
@@ -326,6 +326,31 @@ begin
   { apply_instance }
 end
 
+variables {F} (L)
+
+lemma norm_norm [algebra L F] [is_scalar_tower K L F] [finite_dimensional K F] [is_separable K F]
+  (x : F) :
+  norm K (norm L x) = norm K x :=
+begin
+  let A := algebraic_closure K,
+  apply (algebra_map K A).injective,
+  haveI : finite_dimensional L F := finite_dimensional.right K L F,
+  haveI : finite_dimensional K L := finite_dimensional.left K L F,
+  haveI : is_separable K L := is_separable_tower_bot_of_is_separable K L F,
+  haveI : is_separable L F := is_separable_tower_top_of_is_separable K L F,
+  letI : ∀ (σ : L →ₐ[K] A), by haveI := σ.to_ring_hom.to_algebra; exact fintype (F →ₐ[L] A) :=
```

Comment:
I pushed some tweaks to make this `σ.to_ring_hom.to_algebra` stuff slightly more readable; though I think it would be nice at some point to come up with a better way to write it.

Reply:
should that second one be a `by haveI` or a `by letI`?

Reply:
`have` is better here as it doesn't introduce a `let` binding. The "have forgets things" problem only lasts for the scope of the `by`.